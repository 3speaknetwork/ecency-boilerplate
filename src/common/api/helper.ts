import isElectron from "../util/is-electron";

import defaults from "../constants/defaults.json";

export const apiBase = (endpoint: string): string => {
    let base = '';

    if (isElectron()) {
        if (process.env.NODE_ENV === "development") {
            base = "http://localhost:3000";
        } else {
            base = defaults.base;
        }
    }

    return `${base}${endpoint}`;
}
