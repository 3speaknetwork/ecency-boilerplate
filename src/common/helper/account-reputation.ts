export default (input: string | number): number => {

  if(typeof input === 'string'){
    input = Number(input);
  }
  
  if (isFloat(input)) {
    return Math.floor(input);
  }
  
  if (input === 0) {
    return 25;
  }

  if (!input) {
    return input;
  }

  let neg = false;

  if (input < 0) neg = true;

  let reputationLevel = Math.log10(Math.abs(input));
  reputationLevel = Math.max(reputationLevel - 9, 0);

  if (reputationLevel < 0) reputationLevel = 0;

  if (neg) reputationLevel *= -1;

  reputationLevel = reputationLevel * 9 + 25;

  return Math.floor(reputationLevel);
};

function isFloat(n: string | number) {
  return Number(n) === n && n % 1 !== 0;
}
