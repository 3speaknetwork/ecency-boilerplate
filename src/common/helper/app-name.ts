interface ObjInput {
    name?: string
}

export default (input: string | null | undefined | ObjInput) => {
    if (!input) {
        return '';
    }

    if (typeof input === 'string') {
        return input;
    }

    if (typeof input === 'object' && input.name !== undefined) {
        return input.name;
    }

    return '';
};
