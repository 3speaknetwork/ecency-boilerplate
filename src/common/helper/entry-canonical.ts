import {Entry} from "../store/entries/types";

import apps from "@hiveio/hivescript/apps.json";

import defaults from "../../common/constants/defaults.json";

export default (entry: Entry): string | null => {
    if (entry.json_metadata?.canonical_url) {
        return entry.json_metadata?.canonical_url;
    }

    let scheme = `${defaults.base}/{category}/@{username}/{permlink}`;

    if (entry.json_metadata?.app) {
        const identifier = entry.json_metadata.app.split("/")[0];

        if (apps[identifier]) {
            scheme = apps[identifier].url_scheme;
        }
    }

    if (!scheme) {
        return null;
    }

    return scheme
        .replace("{category}", entry.category)
        .replace("{username}", entry.author)
        .replace("{permlink}", entry.permlink);
};
