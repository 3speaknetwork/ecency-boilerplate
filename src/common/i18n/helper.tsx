import React from "react";

import {_t} from "./index";

interface Props {
    k: string,
    args: {},
    children: JSX.Element
}

export class Tsx extends React.Component<Props> {
    render() {
        const {k, args} = this.props;
        const {children} = this.props;
        return React.cloneElement(children, {dangerouslySetInnerHTML: {__html: _t(k, args)}});
    }
}
