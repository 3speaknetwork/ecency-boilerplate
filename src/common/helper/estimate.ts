import { votingPower } from "../api/hive";
import { ActiveUser } from "../store/active-user/types";
import { DynamicProps } from "../store/dynamic-props/types";
import { Entry } from "../store/entries/types";
import parseAsset from "./parse-asset";

export const estimate = (
  entry: Entry,
  activeUser: ActiveUser,
  dynamicProps: DynamicProps,
  percent: number
): number => {
  if (!activeUser) {
    return 0;
  }

  const { fundRecentClaims, fundRewardBalance, base, quote } = dynamicProps;
  const { data: account } = activeUser;

  if (!account.__loaded) {
    return 0;
  }

  const sign = percent < 0 ? -1 : 1;
  const postRshares = entry.net_rshares;

  const totalVests =
    parseAsset(account.vesting_shares).amount +
    parseAsset(account.received_vesting_shares).amount -
    parseAsset(account.delegated_vesting_shares).amount;

  const userVestingShares = totalVests * 1e6;

  const userVotingPower = votingPower(account) * Math.abs(percent);
  const voteEffectiveShares = userVestingShares * (userVotingPower / 10000) * 0.02;

  // reward curve algorithm (no idea whats going on here)
  const CURVE_CONSTANT = 2000000000000;
  const CURVE_CONSTANT_X4 = 4 * CURVE_CONSTANT;
  const SQUARED_CURVE_CONSTANT = CURVE_CONSTANT * CURVE_CONSTANT;

  const postRsharesNormalized = postRshares + CURVE_CONSTANT;
  const postRsharesAfterVoteNormalized = postRshares + voteEffectiveShares + CURVE_CONSTANT;
  const postRsharesCurve =
    (postRsharesNormalized * postRsharesNormalized - SQUARED_CURVE_CONSTANT) /
    (postRshares + CURVE_CONSTANT_X4);
  const postRsharesCurveAfterVote =
    (postRsharesAfterVoteNormalized * postRsharesAfterVoteNormalized - SQUARED_CURVE_CONSTANT) /
    (postRshares + voteEffectiveShares + CURVE_CONSTANT_X4);

  const voteClaim = postRsharesCurveAfterVote - postRsharesCurve;

  const proportion = voteClaim / fundRecentClaims;
  const fullVote = proportion * fundRewardBalance;

  const voteValue = fullVote * (base / quote);

  if (sign > 0) {
    return Math.max(voteValue * sign, 0);
  }

  return voteValue * sign;
};
