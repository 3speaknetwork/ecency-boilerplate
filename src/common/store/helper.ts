import {Store} from "redux";

const getSymbolFromCurrency = require("currency-symbol-map");

import {AppState} from "./index";
import {ActiveUser, UserPoints} from "./active-user/types";
import {loginAct as loginActiveUser, updateAct as updateActiveUserAct} from "./active-user";

import {getAccount, getDynamicProps} from "../api/hive";
import {getPoints, usrActivity, getPromotedEntries} from "../api/private";
import {reloadAct as reloadUsers} from "./users";
import {reloadAct as reloadReblogs} from "./reblogs";
import {fetchedAct as loadDynamicProps} from "./dynamic-props";
import {fetchedAct as entriesFetchedAct} from "../../common/store/entries";
import {setCurrencyAct as setCurrency, muteNotificationsAct as muteNotifications} from "./global";

import {getCurrencyRate} from "../api/misc";

import currencies from "../constants/currencies.json";

import * as ls from "../../common/util/local-storage";

export const activeUserMaker = (name: string, points: string = "0.000", uPoints: string = "0.000"): ActiveUser => {
    return {
        username: name,
        data: {name},
        points: {
            points,
            uPoints
        }
    }
}

export const activeUserUpdater = async (store: Store<AppState>) => {
    const state = store.getState();
    if (state.activeUser) {
        const {username} = state.activeUser;

        let account;
        try {
            account = await getAccount(username);
        } catch (e) {
            return;
        }

        let points: UserPoints;
        try {
            const p = await getPoints(username);
            points = {points: p.points, uPoints: p.unclaimed_points};
        } catch (e) {
            points = {
                points: "0.000",
                uPoints: "0.000"
            };
        }

        store.dispatch(updateActiveUserAct(account, points));
    }
};

export const clientStoreTasks = (store: Store<AppState>) => {
    // Initial state from browser's local storage
    store.dispatch(reloadUsers());
    store.dispatch(loginActiveUser());
    store.dispatch(reloadReblogs());

    // Load dynamic props
    getDynamicProps().then((resp) => {
        store.dispatch(loadDynamicProps(resp));
    });

    // Update active user in interval
    activeUserUpdater(store).then();
    setInterval(() => {
        activeUserUpdater(store).then();
    }, 60 * 1000);

    // Do check-in in interval
    const checkIn = () => {
        const state = store.getState();
        if (state.activeUser) {
            usrActivity(state.activeUser?.username!, 10).then();
        }
    }
    checkIn();
    setInterval(checkIn, 1000 * 60 * 15 + 8);

    // Inject / update promoted entries to store
    const promotedEntries = () => {
        getPromotedEntries().then(r => {
            store.dispatch(entriesFetchedAct("__promoted__", r, false));
        })
    }
    promotedEntries();
    setInterval(promotedEntries, 1000 * 60 * 5);

    // Currency
    const currency = ls.get("currency");
    if (currency && currencies.find(x => x.id === currency)) {
        const symbol = getSymbolFromCurrency(currency);
        getCurrencyRate(currency).then(rate => {
            store.dispatch(setCurrency(currency, rate, symbol));
        });
    }

    // Notifications
    if (ls.get("notifications") === false) {
        store.dispatch(muteNotifications());
    }
}
