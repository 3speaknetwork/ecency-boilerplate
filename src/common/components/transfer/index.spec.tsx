import React from "react";

import {Transfer, TransferAsset, TransferMode} from "./index";

import {globalInstance, dynamicPropsIntance1} from "../../helper/test-helper";

import TestRenderer from "react-test-renderer";

const defProps = {
    global: globalInstance,
    dynamicProps: dynamicPropsIntance1,
    users: [],
    activeUser: {
        username: 'foo',
        data: {
            name: 'foo',
            balance: '12.234 HIVE',
            sbd_balance: '4321.212',
            savings_balance: '2123.000 HIVE'
        },
        points: {
            points: "0.000",
            uPoints: "0.000"
        }
    },
    transactions: {
        list: [],
        loading: false,
        error: false
    },
    signingKey: '',
    addAccount: () => {
    },
    updateActiveUser: () => {
    },
    setSigningKey: () => {

    },
    onHide: () => {
    }
};

describe('(1) Transfer', () => {
    const mode: TransferMode = 'transfer';
    const asset: TransferAsset = 'HIVE';

    const props = {
        mode,
        asset,
        ...defProps
    };

    const component = TestRenderer.create(<Transfer {...props} />);
    const instance: any = component.getInstance();

    it("(1) Step 1", () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("(2) Step 2", () => {
        instance.setState({step: 2, to: 'bar'});
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("(3) Step 3", () => {
        instance.setState({step: 3});
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("(4) Step 4", () => {
        instance.setState({step: 4});
        expect(component.toJSON()).toMatchSnapshot();
    });
})

describe('(2) Transfer to savings', () => {
    const mode: TransferMode = 'transfer-saving';
    const asset: TransferAsset = 'HBD';

    const props = {
        mode,
        asset,
        ...defProps
    };

    const component = TestRenderer.create(<Transfer {...props} />);
    const instance: any = component.getInstance();

    it("(1) Step 1", () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})


describe('(3) Convert', () => {
    const mode: TransferMode = 'convert';
    const asset: TransferAsset = 'HBD';

    const props = {
        mode,
        asset,
        ...defProps
    };

    const component = TestRenderer.create(<Transfer {...props} />);
    const instance: any = component.getInstance();

    it("(1) Step 1", () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})


describe('(4) Withdraw saving', () => {
    const mode: TransferMode = 'withdraw-saving';
    const asset: TransferAsset = 'HIVE';

    const props = {
        mode,
        asset,
        ...defProps
    };

    const component = TestRenderer.create(<Transfer {...props} />);
    const instance: any = component.getInstance();

    it("(1) Step 1", () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})


describe('(5) Power up', () => {
    const mode: TransferMode = 'power-up';
    const asset: TransferAsset = 'HIVE';

    const props = {
        mode,
        asset,
        ...defProps
    };

    const component = TestRenderer.create(<Transfer {...props} />);
    const instance: any = component.getInstance();

    it("(1) Step 1", () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})
