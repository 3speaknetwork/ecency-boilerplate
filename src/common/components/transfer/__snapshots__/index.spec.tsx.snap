// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Transfer (1) Step 1 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box "
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        1
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Transfer to Account
        </div>
        <div
          className="sub-title"
        >
          Move funds to another Hive account
        </div>
      </div>
    </div>
    <form
      className="transfer-box-body"
    >
      <div
        className="transfer-form"
      >
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            From
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                className="form-control"
                readOnly={true}
                value="foo"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            To
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="suggestion"
            >
              <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                  >
                    @
                  </span>
                </div>
                <input
                  autoFocus={true}
                  className="form-control"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        
        
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Amount
          </label>
          <div
            className="d-flex align-items-center col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                autoFocus={false}
                className="form-control"
                onChange={[Function]}
                placeholder="amount"
                type="text"
                value="0.001"
              />
            </div>
            <div
              className="asset-switch"
            >
              <a
                className="asset selected"
                onClick={[Function]}
              >
                HIVE
              </a>
              <a
                className="asset "
                onClick={[Function]}
              >
                HBD
              </a>
            </div>
          </div>
        </div>
        
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <div
              className="balance"
            >
              Balance
              : 
              <span
                className="balance-num"
                onClick={[Function]}
              >
                12.234
                 
                HIVE
              </span>
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Memo
          </label>
          <div
            className="col-sm-10"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="enter your notes here"
              value=""
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <small
              className="text-muted tr-form-text form-text"
            >
              This memo is public
            </small>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <div
            className="col-sm-10 offset-sm-2"
          >
            <button
              className="btn btn-primary"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`(1) Transfer (2) Step 2 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box"
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        2
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Confirm your transaction
        </div>
        <div
          className="sub-title"
        >
          Please, check if everything is correct
        </div>
      </div>
    </div>
    <div
      className="transfer-box-body"
    >
      <div
        className="confirmation"
      >
        <div
          className="users"
        >
          <div
            className="from-user"
          >
            <span
              className="user-avatar medium"
              style={
                Object {
                  "backgroundImage": "url(https://images.ecency.com/u/foo/avatar/medium)",
                }
              }
            />
          </div>
          <div
            className="arrow"
          >
            <svg
              className="fa-arrow-right"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            className="to-user"
          >
            <span
              className="user-avatar medium"
              style={
                Object {
                  "backgroundImage": "url(https://images.ecency.com/u/bar/avatar/medium)",
                }
              }
            />
          </div>
        </div>
        <div
          className="amount"
        >
          0.001
           
          HIVE
        </div>
        
      </div>
      <div
        className="d-flex justify-content-center"
      >
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
        <span
          className="hr-6px-btn-spacer"
        />
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`(1) Transfer (3) Step 3 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box"
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        3
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Sign transaction
        </div>
        <div
          className="sub-title"
        >
          Sign your transaction
        </div>
      </div>
    </div>
    <div
      className="transfer-box-body"
    >
      <div
        className="sign-tr"
      >
        <div
          className="input-group"
        >
          <div
            className="input-group-prepend"
          >
            <span
              className="input-group-text"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M7,14A2,2 0 0,1 5,12A2,2 0 0,1 7,10A2,2 0 0,1 9,12A2,2 0 0,1 7,14M12.65,10C11.83,7.67 9.61,6 7,6A6,6 0 0,0 1,12A6,6 0 0,0 7,18C9.61,18 11.83,16.33 12.65,14H17V18H21V14H23V10H12.65Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <input
            autoFocus={true}
            className="form-control"
            onChange={[Function]}
            placeholder="Enter active private key"
            type="password"
            value=""
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Sign
            </button>
          </div>
        </div>
        <div
          className="or-divider"
        >
          OR
        </div>
        <div
          className="hs-sign"
        >
          <button
            className="btn btn-outline-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <img
              alt="hivesigner"
              className="hs-logo"
              src="hive-signer.svg"
            />
             
            Sign with hivesigner
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`(1) Transfer (4) Step 4 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box"
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        4
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Success
        </div>
        <div
          className="sub-title"
        >
          Transaction successful
        </div>
      </div>
    </div>
    <div
      className="transfer-box-body"
    >
      <div
        className="success"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Transferred <span class=\\"amount\\">0.001 HIVE</span> From <span class=\\"from-user\\">foo</span> to <span class=\\"to-user\\">bar</span>",
          }
        }
      />
      <div
        className="d-flex justify-content-center"
      >
        <button
          className="btn btn-outline-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Another transfer
        </button>
        <span
          className="hr-6px-btn-spacer"
        />
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Finish
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`(2) Transfer to savings (1) Step 1 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box "
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        1
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Transfer to Savings
        </div>
        <div
          className="sub-title"
        >
          Protect funds by requiring a 3 day withdraw waiting period
        </div>
      </div>
    </div>
    <form
      className="transfer-box-body"
    >
      <div
        className="transfer-form"
      >
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            From
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                className="form-control"
                readOnly={true}
                value="foo"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            To
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="suggestion"
            >
              <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                  >
                    @
                  </span>
                </div>
                <input
                  autoFocus={false}
                  className="form-control"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value="foo"
                />
              </div>
            </div>
          </div>
        </div>
        
        
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Amount
          </label>
          <div
            className="d-flex align-items-center col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                autoFocus={true}
                className="form-control"
                onChange={[Function]}
                placeholder="amount"
                type="text"
                value="0.001"
              />
            </div>
            <div
              className="asset-switch"
            >
              <a
                className="asset "
                onClick={[Function]}
              >
                HIVE
              </a>
              <a
                className="asset selected"
                onClick={[Function]}
              >
                HBD
              </a>
            </div>
          </div>
        </div>
        
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <div
              className="balance"
            >
              Balance
              : 
              <span
                className="balance-num"
                onClick={[Function]}
              >
                4,321.212
                 
                HBD
              </span>
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Memo
          </label>
          <div
            className="col-sm-10"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="enter your notes here"
              value=""
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <small
              className="text-muted tr-form-text form-text"
            >
              This memo is public
            </small>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <div
            className="col-sm-10 offset-sm-2"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`(3) Convert (1) Step 1 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box "
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        1
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Convert
        </div>
        <div
          className="sub-title"
        >
          Only recommended when HBD is below $1. You will get $1 of HIVE in 3.5 days.
        </div>
      </div>
    </div>
    <form
      className="transfer-box-body"
    >
      <div
        className="transfer-form"
      >
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            From
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                className="form-control"
                readOnly={true}
                value="foo"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Amount
          </label>
          <div
            className="d-flex align-items-center col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                autoFocus={true}
                className="form-control"
                onChange={[Function]}
                placeholder="amount"
                type="text"
                value="0.001"
              />
            </div>
            <div
              className="asset-switch"
            >
              <a
                className="asset "
                onClick={[Function]}
              >
                HIVE
              </a>
              <a
                className="asset selected"
                onClick={[Function]}
              >
                HBD
              </a>
            </div>
          </div>
        </div>
        
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <div
              className="balance"
            >
              Balance
              : 
              <span
                className="balance-num"
                onClick={[Function]}
              >
                4,321.212
                 
                HBD
              </span>
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <div
            className="col-sm-10 offset-sm-2"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`(4) Withdraw saving (1) Step 1 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box "
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        1
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Savings Withdraw
        </div>
        <div
          className="sub-title"
        >
          Withdraw funds after the required 3 day waiting period
        </div>
      </div>
    </div>
    <form
      className="transfer-box-body"
    >
      <div
        className="transfer-form"
      >
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            From
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                className="form-control"
                readOnly={true}
                value="foo"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            To
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="suggestion"
            >
              <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                  >
                    @
                  </span>
                </div>
                <input
                  autoFocus={false}
                  className="form-control"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value="foo"
                />
              </div>
            </div>
          </div>
        </div>
        
        
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Amount
          </label>
          <div
            className="d-flex align-items-center col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                autoFocus={true}
                className="form-control"
                onChange={[Function]}
                placeholder="amount"
                type="text"
                value="0.001"
              />
            </div>
            <div
              className="asset-switch"
            >
              <a
                className="asset selected"
                onClick={[Function]}
              >
                HIVE
              </a>
              <a
                className="asset "
                onClick={[Function]}
              >
                HBD
              </a>
            </div>
          </div>
        </div>
        
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <div
              className="balance"
            >
              Balance
              : 
              <span
                className="balance-num"
                onClick={[Function]}
              >
                2,123.000
                 
                HIVE
              </span>
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Memo
          </label>
          <div
            className="col-sm-10"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="enter your notes here"
              value=""
            />
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <small
              className="text-muted tr-form-text form-text"
            >
              This memo is public
            </small>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <div
            className="col-sm-10 offset-sm-2"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`(5) Power up (1) Step 1 1`] = `
<div
  className="transfer-dialog-content"
>
  <div
    className="transfer-box "
  >
    <div
      className="transfer-box-header"
    >
      <div
        className="step-no"
      >
        1
      </div>
      <div
        className="box-titles"
      >
        <div
          className="main-title"
        >
          Stake Hive as Power
        </div>
        <div
          className="sub-title"
        >
          Influence tokens which give you more control over post payouts and allow you to earn on curation rewards.
        </div>
      </div>
    </div>
    <form
      className="transfer-box-body"
    >
      <div
        className="transfer-form"
      >
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            From
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                className="form-control"
                readOnly={true}
                value="foo"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            To
          </label>
          <div
            className="col-sm-10"
          >
            <div
              className="suggestion"
            >
              <div
                className="input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                  >
                    @
                  </span>
                </div>
                <input
                  autoFocus={false}
                  className="form-control"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value="foo"
                />
              </div>
            </div>
          </div>
        </div>
        
        
        <div
          className="form-group row"
        >
          <label
            className="form-label col-form-label col-sm-2"
          >
            Amount
          </label>
          <div
            className="d-flex align-items-center col-sm-10"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  @
                </span>
              </div>
              <input
                autoFocus={true}
                className="form-control"
                onChange={[Function]}
                placeholder="amount"
                type="text"
                value="0.001"
              />
            </div>
          </div>
        </div>
        
        <div
          className="row"
        >
          <div
            className="col-md-10 offset-md-2"
          >
            <div
              className="balance"
            >
              Balance
              : 
              <span
                className="balance-num"
                onClick={[Function]}
              >
                12.234
                 
                HIVE
              </span>
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <div
            className="col-sm-10 offset-sm-2"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
